modules/repository/create.go

```diff
@@ -44,39 +46,77 @@ func CreateRepository(doer, u *models.User, opts models.CreateRepoOptions) (_ *m
 		TrustModel:                      opts.TrustModel,
 	}
 
-	err = models.WithTx(func(ctx models.DBContext) error {
-		if err = models.CreateRepository(ctx, doer, u, repo); err != nil {
+	if err := models.WithTx(func(ctx models.DBContext) error {
+		if err := models.CreateRepository(ctx, doer, u, repo, false); err != nil {
 			return err
 		}
 
 		// No need for init mirror.
-		if !opts.IsMirror {
-			repoPath := models.RepoPath(u.Name, repo.Name)
-			if err = initRepository(ctx, repoPath, doer, repo, opts); err != nil {
-				if err2 := util.RemoveAll(repoPath); err2 != nil {
-					log.Error("initRepository: %v", err)
-					return fmt.Errorf(
-						"delete repo directory %s/%s failed(2): %v", u.Name, repo.Name, err2)
-				}
-				return fmt.Errorf("initRepository: %v", err)
+		if opts.IsMirror {
+			return nil
+		}
+
+		repoPath := models.RepoPath(u.Name, repo.Name)
+		if com.IsExist(repoPath) {
+			// repo already exists - We have two or three options.
+			// 1. We fail stating that the directory exists
+			// 2. We create the db repository to go with this data and adopt the git repo
+			// 3. We delete it and start afresh
+			//
+			// Previously Gitea would just delete and start afresh - this was naughty.
+			// if opts.AdoptPreExisting {
```

Comment:
This commented out code should be deleted

Reply:
Sorry missed that.